extension/controller/translation/translationWorker.js

```diff
@@ -246,7 +288,7 @@ class TranslationHelper {
         // instantiate the Translation Service
         constructTranslationService() {
             if (!this.translationService) {
-                let translationServiceConfig = {};
+                let translationServiceConfig = { cacheSize: 10 };
```

Comment:
Just a note: a cache size of 10 won't do you much good. The cache is non-probing, [it is basically just `cache[hash(sentence) % cacheSize]`](https://github.com/browsermt/bergamot-translator/blob/22d6bc07e7a6e83bc22676a106fe0ae8fc691c15/src/translator/cache.h#L65-L70). The [description of the parameter](https://github.com/browsermt/bergamot-translator/blob/22d6bc07e7a6e83bc22676a106fe0ae8fc691c15/src/translator/service.h#L33-L37) is a bit indirect about that. If you set it too low, you'll end up having too many different sentences hitting the same cache entry, constantly overwriting each other, and no cache benefit at all.

In my experience you'd get about 20% occupancy, so if you set it to 50 you'd be caching about 10 sentences. But from testing in TranslateLocally and my extension fork, I'd suggest starting with something around 1000 or higher, and see whether you can notice it in the memory usage.

Reply:
Thanks for the tip Jelmer, please keep them coming. I think caching will be particularly helpful after I move the engine to the background script last week. I heard from the security folks that's fine, so I think that will be helpful.